### DISCUSSION TITLE
Frigate NVR’s out of box amateurishness  -default Auto-Purge to Prevent Storage Failure???

### DISCUSSION THREAD

**Discussion by 3mandd** [blakeblackshear/frigate#14819] [Ask A Question] [answered]
It's mind-boggling that Frigate NVR is designed to fill up storage with no default limits, warnings, or automatic cleanup. Users are left to manually configure retention settings, making it nearly guaranteed to fail on limited storage systems. For a surveillance tool where reliability is key, this lack of storage management is a glaring oversight that leaves users in the lurch unless they anticipate the issue themselves.

Why not implement a default auto-purge setting? This feature would automatically delete the oldest footage when storage reaches a certain threshold (e.g., 80% full), preventing crashes without relying on users to configure it manually.

**Comment by moandcompany** [comment]
You get more than what you paid for :)

**Comment by hawkeye217** [comment]
You should check out the official docs. https://docs.frigate.video/configuration/record/#will-frigate-delete-old-recordings-if-my-storage-runs-out

**Reply by 3mandd** [reply]
> You should check out the official docs. https://docs.frigate.video/configuration/record/#will-frigate-delete-old-recordings-if-my-storage-runs-out

are you saying that Frigate NVR  will not eventually fill up storage and fail by default out of the box?

**Reply by hawkeye217** [reply, marked as answer]
> Will Frigate delete old recordings if my storage runs out?
> As of Frigate 0.12 if there is less than an hour left of storage, the oldest 2 hours of recordings will be deleted.

**Reply by 3mandd** [reply]
> > Will Frigate delete old recordings if my storage runs out?
> > As of Frigate 0.12 if there is less than an hour left of storage, the oldest 2 hours of recordings will be deleted.

ok thanks, will investigate more.  It (0.14.1-f4f3cfa) failed to do that running in docker on bookworm.

**Reply by hawkeye217** [reply]
If you can reproduce something you feel is an issue, go ahead and open a support discussion and we'd be happy to take a look.
